mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -336,14 +361,19 @@ class GeneralizedLinearRegression @Since("2.0.0") (@Since("2.0.0") override val
     }
 
     val w = if (!isDefined(weightCol) || $(weightCol).isEmpty) lit(1.0) else col($(weightCol))
-    val instances: RDD[Instance] =
-      dataset.select(col($(labelCol)), w, col($(featuresCol))).rdd.map {
-        case Row(label: Double, weight: Double, features: Vector) =>
-          Instance(label, weight, features)
-      }
+    val off = if (!isDefined(offsetCol) || $(offsetCol).isEmpty) {
```

Comment:
maybe rename off to offset?  also it looks like the spacing below needs to be fixed (extra indentation).
+    val off = if (!isDefined(offsetCol) || $(offsetCol).isEmpty) { 
+      lit(0.0) 
+    } else { 
+      col($(offsetCol)).cast(DoubleType) 
+    }